script.py

```diff
@@ -0,0 +1,239 @@
+    )
+
+
+def render_html_report(event):
+    return template.render(
+        name=event["name"],
+        start_time=event["name"],
+        event_id=event["name"],
+        charts=[
+            {
+                "chart_type": "bar",
+                "x": [city for (city, _) in city_client_info],
+                "y": [data for (_, data) in city_client_info],
+                "title": "Unique Clients Per City",
+            },
+            {
+                "chart_type": "bar",
+                "x": [city for (city, _) in city_ip_info],
+                "y": [data for (_, data) in city_ip_info],
+                "title": "Unique IPs Per City",
+            },
+            {
+                "chart_type": "bar",
+                "x": [resolution for (resolution, _) in resolution_client_info],
+                "y": [data for (_, data) in resolution_client_info],
+                "title": "Unique Clients Per Resolution",
+            },
+            {
+                "chart_type": "bar",
+                "x": [os for (os, _) in os_client_info],
+                "y": [data for (_, data) in os_client_info],
+                "title": "Unique Clients Per OS",
+            },
+            {
+                "chart_type": "bar",
+                "x": [browser for (browser, _) in browser_client_info],
+                "y": [data for (_, data) in browser_client_info],
+                "title": "Unique Clients Per Browser",
+            },
+            {
+                "chart_type": "histogram",
+                "x": watch_times,
+                "y": "",
+                "title": "Watch Times (mins)",
+            },
+        ],
+        distinct_ips=len(set(ips)),
+        distinct_clients=len(set(client_ids)),
+    )
+
+
+auth_request = requests.post(
+    "https://central.livingasone.com/api/v3/login?newToken=true",
+    json={"userName": USERNAME, "password": PASSWORD,},
+)
+
+customer_id = auth_request.json()["customerId"]
+events_request = requests.get(
+    f"https://central.livingasone.com/api/v3/customers/{customer_id}/webevents",
+    cookies=auth_request.cookies,
+)
+
+
+if FILENAME in os.listdir(DIR):
+    with open(os.path.join(DIR, FILENAME), "r") as f:
+        data = json.loads(f.read())
+else:
+    data = {"events": []}
+
+new_uuids = []
+for event in [e for e in events_request.json()]:
+    uuid = event["uuid"]
+    if uuid in [e["event_id"] for e in data["events"]]:
+        continue
+    new_uuids.append(uuid)
+    event_data = {}
+    event_data["event_id"] = uuid
+    event_data["start_time"] = event["startTime"]
+    event_data["name"] = event["name"]
+
+    public_info = requests.get(
+        f"https://central.livingasone.com/api_v2.svc/public/events/{uuid}/status/rep?geoData=true"
+    )
+    event_data["public_info"] = public_info.json()
+
+    detailed_info = requests.get(
+        f"https://central.livingasone.com/api/v3/customers/{customer_id}/webevents/{uuid}/export?max=500",
```

Comment:
What happens without the `max` param? I'm guessing that their dashboard uses this also, but it seems like a good caveat to note in the readme for folks,  so they're aware of the potential data truncation/loss.

Reply:
No idea what happens without the max param and if it will just be implicitly enforced. Will call that out.